test/sdk.test.ts

```diff
@@ -1,45 +1,81 @@
+import { BrowserOptions } from '@sentry/browser';
+
 import { CapacitorOptions } from '../src';
-import { DEFAULT_OPTIONS_MOBILE, DEFAULT_OPTIONS_WEB, init } from '../src/sdk';
+import { init } from '../src/sdk';
 import { NATIVE } from '../src/wrapper';
 
 jest.mock('../src/wrapper', () => {
   return {
     NATIVE: {
       platform: 'web',
-      initNativeSdk: jest.fn(),
+      initNativeSdk: jest.fn(() => Promise.resolve()),
     },
   };
 });
 
-describe('SDK Init', () => {
-  it('Uses default web options on web', () => {
-    NATIVE.platform = 'web';
+beforeEach(() => {
+  jest.clearAllMocks();
+});
 
-    init({ dsn: '' }, (finalOptions: CapacitorOptions) => {
-      expect(finalOptions.enableNative).toBe(false);
-      expect(finalOptions).toEqual(
-        expect.objectContaining(DEFAULT_OPTIONS_WEB),
-      );
-    });
-  });
+describe('SDK Init', () => {
+  // [string, platform, options, autoSessionTracking, enableNative, enableAutoSessionTracking]
+  const table: Array<[
+    string,
+    string,
+    CapacitorOptions & BrowserOptions,
+    boolean,
+    boolean,
+    boolean,
+  ]> = [
+    ['Uses default options on web', 'web', { dsn: '' }, true, false, false],
+    ['Uses default options on ios', 'ios', { dsn: '' }, false, true, true],
+    [
+      'Uses default options on android',
+      'android',
+      { dsn: '' },
+      false,
+      true,
+      true,
+    ],
+    [
+      'enableAutoSessionTracking sets autoSessionTracking on web',
+      'web',
+      { dsn: '', enableAutoSessionTracking: false },
+      false,
+      false,
+      false,
+    ],
+    [
+      'enableAutoSessionTracking sets enableAutoSessionTracking on android',
+      'android',
+      { dsn: '', enableAutoSessionTracking: false },
+      false,
+      true,
+      false,
+    ],
+    [
+      'enableAutoSessionTracking sets enableAutoSessionTracking on ios',
+      'ios',
+      { dsn: '', enableAutoSessionTracking: false },
+      false,
+      true,
+      false,
```

Comment:
Let’s make these last two booleans -> the object containing enableNative and enableAutoSessionTracking